Parser/ABOParser.php

```diff
@@ -204,53 +179,223 @@ protected function parseTransactionLine($line)
         $transaction = $this->getTransactionClass();
 
         # Receipt ID
-        $receiptId = ltrim(substr($line, 35, 13), '0');
-        $transaction->setReceiptId($receiptId);
+        $transaction->setReceiptId($this->parseReceiptId($line));
 
         # Debit / Credit
-        $amount = ltrim(substr($line, 48, 12), '0') / 100;
-        $postingCode = substr($line, 60, 1);
-        switch ($postingCode) {
+        switch ($this->parsePostingCode($line)) {
             case self::POSTING_CODE_DEBIT:
-                $transaction->setDebit($amount);
+                $transaction->setDebit($this->parseAmount($line));
```

Comment:
I would suggest to create something like `$parsedAmount = $this->parseAmount($line);` before `switch` and use this variable then, because the same call is done all the time (4-times).